pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2307,4 +2253,16 @@ private boolean isSystemTopic(String topic) {
     public void setInterceptor(BrokerInterceptor interceptor) {
         this.interceptor = interceptor;
     }
+
+    public void pausedConnections(int numberOfConnections) {
+        pausedConnections.addAndGet(numberOfConnections);
```

Comment:
Is this exported via prometheus? It would be better to have separate event counters for pause and resume so that if pausing happens between prometheus pulls we can see it.

Reply:
Good point. I just exposed it here to for validation in the tests, though it makes sense to expose the 2 counters.